src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -315,7 +336,9 @@
         (DELETE [""] [subject]
           (handle-delete-certificate-request! subject ca-settings)))
       (GET ["/certificate_revocation_list/" :ignored-node-name] []
-        (handle-get-certificate-revocation-list ca-settings)))
+        (handle-get-certificate-revocation-list ca-settings))
+      (PUT ["/clean"] request
```

Comment:
Would it work/make senes to do something like `[subject :as {body :body}]` like we do above, in the `certificate_request` endpoint? Mostly I'm just not seeing this `request` pattern elsewhere, so it's a little confusing to me whether that's a totally different thing or not.

Reply:
I was having a hard time getting this to bind right, and didn't want to use "subject" because unlike the other endpoints here, this doesn't act on a single "subject" but rather a list of them, but we can't extract them until after we parse them out of the JSON. So I handle pulling apart the request body in the handler. But maybe I've done something wrong and there is a way to get the destructuring to work? For now, I've followed the pattern we use for other (non-CA) endpoints in puppetserver (see the routes in `master_core`) that process a body withouth going through liberator like the `certificate_status` endpoint does.